### DISCUSSION TITLE
clipping intersecting mesh of lesser importance?

### DISCUSSION THREAD

**Discussion by myociss** [CGAL/cgal#8621] [Q&A] [answered]
I have two intersecting surface meshes representing different materials that I need to repair, with one mesh (mesh A) representing the more important material and the other (mesh B) representing the less important one. If I find all faces of mesh B that intersect mesh A using the Polygon_mesh_processing library, is there some straightforward way to then clip these faces of mesh B so that the region inside mesh A is removed, i.e. some function that will return new faces of mesh B as clipped by mesh A? Thanks so much in advance!

**Comment by sloriot** [comment]
did you try to use the [PMP::clip](https://doc.cgal.org/latest/Polygon_mesh_processing/group__PMP__corefinement__grp.html#ga5dc9e7acf8433235adc7b91ac93a7d42) function? Could you share one such example?

**Reply by myociss** [reply]
Great, this looks very close to what I want, however I would like to keep the part of mesh B that is outside A rather than inside. Is this possible?

**Reply by MaelRL** [reply, marked as answer]
If your meshes are both bounding volumes, you can use  [PMP::corefine_and_compute_difference](https://doc.cgal.org/latest/Polygon_mesh_processing/group__PMP__corefinement__grp.html#ga02fa6ed9116a759ecfb87063392f5455); otherwise, you can use [PMP::split](https://doc.cgal.org/latest/Polygon_mesh_processing/group__PMP__corefinement__grp.html#ga642520055888a41a3e1a76965a1f9fbb) and identify the connected component(s) that you must discard by checking if it's within A using (for example) [CGAL::Side_of_triangle_mesh](https://doc.cgal.org/latest/Polygon_mesh_processing/classCGAL_1_1Side__of__triangle__mesh.html).

**Reply by myociss** [reply]
Oh okay, it looks like you're saying split() will divide mesh B into components that are completely inside of or completely outside of mesh A, and then I can simply discard the components that are inside mesh A. If so, that will work perfectly, thank you!
